### PR TITLE
New addon: registry-aliases

### LINKED ISSUE #6656: Addon to configure internal registry aliases
Based on the discussion on #4604 , we need an addon that can configure the registry aliases. 

The addon will do the following:

- Deploy a ConfigMap that can define the aliases
- Deploy a Daemonset that can update the minikube node /etc/hosts 
- Deploy a Job that can patch the coreDNS for aliases rewrite rules
- SA and ClusterRoleBinding to run the  Job

### PR DESCRIPTION
**Pull request by kameshsampath**
- Service Account and binding to run the job
- Registry aliases ConfigMap
- Registry aliases daemonset to update the node etc/hosts

fixes #6656 

Signed-off-by: Kamesh Sampath <ksampath@redhat.com>

### CODE REVIEW

**Comment by medyagh** [comment]
Do you mind putting some examples of using this ?

**Comment by kameshsampath** [comment]
yes I can do that I have working example folder which users can refer to

**Comment by kameshsampath** [comment]
@medyagh and @tstromberg  - the PR is updated with README and usage.  Please review and merge

We need to handle the reversal of things done by add-on, we shall take that as separate issue .

File: deploy/addons/registry-aliases/README.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Minikube Registry Aliases Addon
+
+An addon to minikube that can help push and pull from the minikube registry using custom domain names. The custom domain names will be made resolveable from with in cluster and at minikube node.
+
+## How to use ?
+
+### Start minikube
+
+```shell
+minikube profile demo
```
[the remark this answers is not in the document]
Comment (kameshsampath) [reply deploy/addons/registry-aliases/README.md]: Done thanks ! @medyagh ^^

**Comment by tstromberg** [comment]
Thank you!

**Comment by kameshsampath** [comment]
Thank you for merging it!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**